### PR TITLE
cloud: add release notes for 20231017

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

Add release notes for Release 20231017:

- https://github.com/pingcap/docs/pull/15031
- https://github.com/pingcap/docs/pull/15033

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/tidb-cloud-release-notes.md [hunk 1]
```diff
@@ -8,6 +8,25 @@ aliases: ['/tidbcloud/supported-tidb-versions','/tidbcloud/release-notes']
 
 This page lists the release notes of [TiDB Cloud](https://www.pingcap.com/tidb-cloud/) in 2023.
 
+## October 17, 2023
+
+**General changes**
+
+- [Data Service (beta)](https://tidbcloud.com/console/data-service) provides a sample Data App to help you get started quickly.
+
+    If your project does not have any Data App yet, you can follow the on-screen instructions on the [Data Service](https://tidbcloud.com/console/data-service) page to create a sample Data App with a few clicks. This feature provides a smooth onboarding experience and helps you explore Data Service features.
```
Comment (Oreoxmt) [inline tidb-cloud/tidb-cloud-release-notes.md:17]:
```suggestion
    If your project does not have any Data App yet, you can follow the on-screen instructions on the [**Data Service**](https://tidbcloud.com/console/data-service) page to create a sample Data App with a few clicks. This feature provides a smooth onboarding experience and helps you explore Data Service features.
```

File: tidb-cloud/tidb-cloud-release-notes.md
Comment (Oreoxmt) [inline tidb-cloud/tidb-cloud-release-notes.md:1]: Preview: https://scaffold-git-preview-cloud-pingcapdocs15107-oreoxmt.vercel.app/tidbcloud/tidb-cloud-release-notes#october-17-2023

File: tidb-cloud/tidb-cloud-release-notes.md [hunk 2]
```diff
@@ -8,6 +8,25 @@ aliases: ['/tidbcloud/supported-tidb-versions','/tidbcloud/release-notes']
...
+
+- [Data Service (beta)](https://tidbcloud.com/console/data-service) provides a sample Data App to help you get started quickly.
+
+    If your project does not have any Data App yet, you can follow the on-screen instructions on the [**Data Service**](https://tidbcloud.com/console/data-service) page to create a sample Data App with a few clicks. This feature provides a smooth onboarding experience and helps you explore Data Service features.
+
+    For more information, see [Get started with a sample Data App](/tidb-cloud/data-service-get-started.md#get-started-with-a-sample-data-app).
+
+- [Data Service (beta)](https://tidbcloud.com/console/data-service) supports array type parameters and introduces an **Enum** property for parameters.
+
+    - The array type parameters facilitate you to perform batch operations, such as using `IN` to match multiple records in a `WHERE` statement.
```
Comment (qiancai) [inline tidb-cloud/tidb-cloud-release-notes.md:23]:
```suggestion
    - The array type parameters facilitate you to perform batch operations, such as using `IN` in a `WHERE` statement to match multiple records.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Review by hfxsd — APPROVED** [review]

**Pull request closed without merging**